File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    num = {"zero":'0', "one":'1', "two":'2', "three":'3', "four":'4', "five":'5', "six":'6', "seven":'7', "eight":'8', "nine":'9'}
    while s.isdigit() != True:
        for n in num:
            if n in s:
                s = s.replace(n, num[n])
    return int(s)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    let answer = s;
    const num = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];

    for(let i = 0; i < num.length; i++) {
        let arr = answer.split(num[i]);
        answer = arr.join(i);
    }
    return Number(answer);
}